MusicTaste/MusicTaste/APIManagers/SpotifyAPIManager.m

```diff
@@ -146,6 +148,28 @@ - (NSDictionary *) convertSpotifyTracks:(NSArray *)tracks{
     //Returns all the data into a dictionary
     return @{@"albums": albumSet, @"artists": artistSet, @"tracks":trackSet};
 }
--(void) saveData:(NSDictionary *) spotifyData{
+-(void) saveSpotifyData:(NSDictionary *) spotifyData{
+    //Saves all Spotify Data such as there top genres,tracks,artist,album,and the artist images to the Parse Database
+    PFUser *current = [PFUser currentUser];
+    current[@"genres"] = [NSArray arrayWithArray:spotifyData[@"genres"]];
+    current[@"tracks"] = [NSArray arrayWithArray:spotifyData[@"tracks"]];
+    current[@"artists"] = [NSArray arrayWithArray:spotifyData[@"artists"]];
+    current[@"albums"] = [NSArray arrayWithArray:spotifyData[@"albums"]];
+    current[@"images"] = [NSArray arrayWithArray:spotifyData[@"images"]];
+    [PFUser.currentUser saveInBackground];
 }
+-(void) saveSpotifyUserData:(NSDictionary *) spotifyData{
+    // Saves name and Profile Picture to User
+    [self getSpotifyData:@"https://api.spotify.com/v1/me" completion:^(NSDictionary * userDict, NSError * error) {
+        if (!error){
+            NSString *name = userDict[@"display_name"];
+            NSArray *image= userDict[@"images"];
```

Comment:
Probably you want to name this `NSArray *images = userDict[@"images"];` since there are multiple